projects/plugins/jetpack/extensions/blocks/nextdoor/edit.js

```diff
@@ -93,23 +93,45 @@ export function NextdoorEdit( {
 		</Placeholder>
 	);
 
-	const blockPreview = () => {
-		return (
-			<>
-				<div className={ `${ defaultClassName }-overlay` }></div>
-				<iframe width="100%" height="200" frameBorder="0" src={ iframeUrl } title="Nextdoor" />
-			</>
-		);
-	};
-
 	return (
 		<div className={ className }>
 			<NextdoorControls
 				{ ...{ defaultClassName, nextdoorShareUrl, onFormSubmit, setNextdoorShareUrl } }
 			/>
-			{ iframeUrl ? blockPreview() : blockPlaceholder }
+			{ iframeUrl ? (
+				<BlockPreview defaultClassName={ defaultClassName } iframeUrl={ iframeUrl } />
+			) : (
+				blockPlaceholder
+			) }
 		</div>
 	);
 }
 
+const BlockPreview = ( { defaultClassName, iframeUrl } ) => {
+	const html = `
+		<script>
+			window.addEventListener( 'message', function ( event ) {
+				if ( ! event.origin.startsWith( 'https://nextdoor' ) ) {
+					return;
+				}
+				var iframe = document.getElementById( 'nextdoor-embed' );
+				if ( event.source !== iframe.contentWindow ) {
+					return;
+				}
+				iframe.setAttribute( 'height', event.data.height );
+				var clientBoundingRect = document.body.getBoundingClientRect();
+				window.parent.postMessage( { action: 'resize', width: clientBoundingRect.width, height: clientBoundingRect.height }, '*' );
+			} );
+		</script>
+		<iframe id="nextdoor-embed" width="100%" height="200" frameBorder="0" src="${ iframeUrl }" title="Nextdoor" />
```

Comment:
What if there are more than one nextdoor embeds on the page? `id` attribute will need to be different.

Reply:
Fixed in https://github.com/Automattic/jetpack/pull/33931/commits/ccf50e7755f19436b44b2a5b4980a7c9407c7cfe